File: app/src/main/java/com/test/ashish/githubissues/IssuesViewModel.java
package com.test.ashish.githubissues;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;
import android.util.Log;

import com.test.ashish.githubissues.Pojo.Issues;

import java.util.List;

public class IssuesViewModel extends AndroidViewModel {

    private IssuesRepository issuesRepository;
    private LiveData<List<Issues>> allIssues;

    public IssuesViewModel(@NonNull Application application) {
        super(application);
        issuesRepository = new IssuesRepository(application);
    }

    public void deleteAll(long time){
        issuesRepository.deleteAllIssues(time);
    }

    public void fetchApi(String orgName, String repoName){
        issuesRepository.fetchApi(orgName, repoName);
    }
    public LiveData<List<Issues>> getAllIssues(String orgname, String repo){
        return issuesRepository.getAllIssues(orgname, repo);
    }

}

File: app/src/main/java/com/test/ashish/githubissues/IssuesViewModel.kt
package com.test.ashish.githubissues

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.test.ashish.githubissues.Pojo.Issues

class IssuesViewModel(application: Application) : AndroidViewModel(application) {
    private val issuesRepository: IssuesRepository
    private val allIssues: LiveData<List<Issues>>? = null
    private var status: MutableLiveData<Boolean>? = null
    fun deleteAll(time: Long) {
        issuesRepository.deleteAllIssues(time)
    }

    fun getAllIssues(
        orgname: String?,
        repo: String?
    ): LiveData<List<Issues?>?>? {
        return issuesRepository.getAllIssues(orgname!!, repo!!)
    }

    val currentStatus: MutableLiveData<Boolean>?
        get() {
            if (status == null) {
                status = issuesRepository.currentStatus
            }
            return status
        }

    init {
        issuesRepository = IssuesRepository(application)
    }
}